watchtower/blob/justice_kit.go

```diff
@@ -1,153 +1,269 @@
+	"github.com/lightningnetwork/lnd/lnwallet"
 	"github.com/lightningnetwork/lnd/lnwire"
 )
 
-// JusticeKit is lé Blob of Justice. The JusticeKit contains information
-// required to construct a justice transaction, that sweeps a remote party's
-// revoked commitment transaction. It supports encryption and decryption using
-// chacha20poly1305, allowing the client to encrypt the contents of the blob,
-// and for a watchtower to later decrypt if action must be taken. The encoding
-// format is versioned to allow future extensions.
-type JusticeKit struct {
-	// BlobType encodes a bitfield that inform the tower of various features
-	// requested by the client when resolving a breach. Examples include
-	// whether the justice transaction contains a reward for the tower, or
-	// whether the channel is a legacy or anchor channel.
-	//
-	// NOTE: This value is not serialized in the encrypted payload. It is
-	// stored separately and added to the JusticeKit after decryption.
-	BlobType Type
-
-	// SweepAddress is the witness program of the output where the client's
-	// fund will be deposited. This value is included in the blobs, as
-	// opposed to the session info, such that the sweep addresses can't be
-	// correlated across sessions and/or towers.
-	//
-	// NOTE: This is chosen to be the length of a maximally sized witness
-	// program.
-	SweepAddress []byte
-
-	// RevocationPubKey is the compressed pubkey that guards the revocation
-	// clause of the remote party's to-local output.
-	RevocationPubKey PubKey
-
-	// LocalDelayPubKey is the compressed pubkey in the to-local script of
-	// the remote party, which guards the path where the remote party
-	// claims their commitment output.
-	LocalDelayPubKey PubKey
-
-	// CSVDelay is the relative timelock in the remote party's to-local
-	// output, which the remote party must wait out before sweeping their
-	// commitment output.
-	CSVDelay uint32
-
-	// CommitToLocalSig is a signature under RevocationPubKey using
-	// SIGHASH_ALL.
-	CommitToLocalSig lnwire.Sig
-
-	// CommitToRemotePubKey is the public key in the to-remote output of the
-	// revoked commitment transaction.
-	//
-	// NOTE: This value is only used if it contains a valid compressed
-	// public key.
-	CommitToRemotePubKey PubKey
-
-	// CommitToRemoteSig is a signature under CommitToRemotePubKey using
-	// SIGHASH_ALL.
-	//
-	// NOTE: This value is only used if CommitToRemotePubKey contains a
-	// valid compressed public key.
-	CommitToRemoteSig lnwire.Sig
+// JusticeKit is an interface that describes lé Blob of Justice. An
+// implementation of the JusticeKit contains information required to construct
+// a justice transaction, that sweeps a remote party's revoked commitment
+// transaction. It supports encryption and decryption using chacha20poly1305,
+// allowing the client to encrypt the contents of the blob, and for a
+// watchtower to later decrypt if action must be taken.
+type JusticeKit interface {
```

Comment:
Nice, I dig the over all approach here of pushing down the commit type specific behavior into the lowest level possible, which hides all but the essential information from the caller.